api/src/main/resources/org/openmrs/module/commonreports/sql/familyPlanning.sql

```diff
@@ -0,0 +1,34 @@
+SELECT
+count(if( concept_id=(select DISTINCT concept_id from concept_name where name = 'FP administred' and locale='en' and concept_name_type='FULLY_SPECIFIED' and voided = 0) && value_coded=(select DISTINCT concept_id from concept_name where name = 'Microgynon (plaquette)' and locale='en' and concept_name_type='FULLY_SPECIFIED' and voided = 0) && obs_group_id IN (select obs_group_id from obs where obs_group_id IN (SELECT obs_id FROM obs where obs_datetime >= DATE_SUB(:startDate, INTERVAL 1 MONTH) AND obs_datetime <= :endDate AND concept_id=(select DISTINCT concept_id from concept_name where name = 'Family Planning' and locale='en' and concept_name_type='FULLY_SPECIFIED' and voided = 0)) AND person.gender = 'F' AND round(DATEDIFF(obs.obs_datetime, person.birthdate)/365.25, 1) < 25 AND concept_id=(select DISTINCT concept_id from concept_name where name = 'Type of user' and locale='en' and concept_name_type='FULLY_SPECIFIED' and voided = 0) AND value_coded=(select DISTINCT concept_id from concept_name where name = 'New' and locale='en' and concept_name_type='FULLY_SPECIFIED' and voided = 0)) ,1,null)) as newMycogynonFemaleLT25,
```

Comment:
Hi @reagan-meant, thanks for this PR. I'm worried of the overhead that this repeated query would cause on the overall report query. Could the queries not be combined into one? It'd also be great providing a dynamic list of concepts than having to hardcode them in the query especially by name which could change anytime in the future.

Reply:
Ideally you'd replace `concept_id=(select .....)` with something like `concept_id=:*ConceptId` where `*` here represents something like `fpAdministred....`. The `*ConceptId`s would be `replace`d somewhere in the code.

Reply:
Done